cmd/apps/rancher_app.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) arkade author(s) 2020. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/k8s"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/alexellis/arkade/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallRancher() *cobra.Command {
+	var rancher = &cobra.Command{
+		Use:          "rancher",
+		Short:        "Install rancher",
+		Long:         "Install rancher to manage kubernetes clusters",
+		Example:      "arkade install rancher --hostname mycluster.example.com ",
+		SilenceUsage: true,
+	}
+
+	rancher.Flags().StringP("namespace", "n", "cattle-system", "The namespace to install rancher")
+	rancher.Flags().Bool("update-repo", true, "Update the helm repo")
+	rancher.Flags().String("hostname", "", "public DNS record to access rancher")
+	rancher.Flags().String("ingress-source", "letsEncrypt", "source of TLS cert: letsEncrypt, rancher, secret")
+	rancher.Flags().String("letsencrypt-email", "", "email address used for communication about your certificate")
+	rancher.Flags().Int("replicas", 3, "Number of replicas of rancher pods")
+	rancher.Flags().String("tls-cert", "", "Path to PEM encoded public key certificate")
+	rancher.Flags().String("tls-key", "", "Path to private key associated with given certificate")
+	rancher.Flags().String("tls-ca", "", "Path to private CA certificate")
+
+	rancher.RunE = func(command *cobra.Command, args []string) error {
+		// Get all flags
+		namespace, _ := command.Flags().GetString("namespace")
+		hostname, _ := command.Flags().GetString("hostname")
+		ingressSource, _ := command.Flags().GetString("ingress-source")
+		letsencryptEmail, _ := command.Flags().GetString("letsencrypt-email")
```

Comment:
Is this required for Rancher to work? How do people run it locally? (inlets-operator could help here if LE is required)

Reply:
`letsencrypt-email` is only required if using LE for ingress, but I didn't check if it actually is a hard requirement on the chart itself.

Locally you could used the `rancher` or `secret` alternatives for `ingress-source`